core/include/gnuradio-4.0/Block.hpp

```diff
@@ -714,11 +715,18 @@ class Block : public lifecycle::StateMachine<Derived> {
     /**
      * Merge tags from all sync ports into one merged tag, apply auto-update parameters
      */
-    constexpr void updateMergedInputTagAndApplySettings(auto& inputSpans, std::size_t untilLocalIndex = 1UZ) noexcept {
+    void updateMergedInputTagAndApplySettings(auto& inputSpans, std::size_t untilLocalIndex = 1UZ) noexcept {
         for_each_reader_span(
             [this, untilLocalIndex](auto& in) {
                 if (in.isSync) {
-                    for (const auto& [key, value] : in.getMergedTag(untilLocalIndex).map) {
+                    std::size_t untilLocalIndex2 = untilLocalIndex;
+                    if constexpr (backwardTagForwarding) {
+                        in.setConsumeOnlyFirstTag(false);
```

Comment:
This is a static information. Why making it a dynamic setting in Port? Should be linked to -- for example -- NTTP parameter in the tag reader.